tools/cell_census_builder/validate.py

```diff
@@ -288,6 +302,35 @@ def count_nonzero(arr: Union[sparse.spmatrix, npt.NDArray[Any]]) -> int:
     return True
 
 
+def _validate_X_layer_has_unique_coords(args: Tuple[str, ExperimentBuilder, str, int, int]) -> bool:
+    """Validate that all X layers have no duplicate coordinates"""
+    soma_path, experiment_builder, layer_name, row_range_start, row_range_stop = args
+    census = soma.Collection(soma_path, context=SOMA_TileDB_Context())
+    census_data = census[CENSUS_DATA_NAME]
+    se = census_data[experiment_builder.name]
+    logging.info(
+        f"validate_no_dups_X start, {experiment_builder.name}, {layer_name}, rows [{row_range_start}, {row_range_stop})"
+    )
+    if layer_name not in se.ms["RNA"].X:
+        return True
+
+    X_layer = se.ms["RNA"].X[layer_name]
+    n_rows, n_cols = X_layer.shape
+    ROW_SLICE_SIZE = 100_000
+
+    for row in range(row_range_start, min(row_range_stop, n_rows), ROW_SLICE_SIZE):
```

Comment:
A few performance Q's:
* How long does this take to run?
* Might we need to parallelize this operation? 
* Is 100K rows a "tuned" value or a WAG? 😄
* AFAIK, we can't avoid retrieving the matrix values themselves w/the current API. I'm not missing something right?

Reply:
It adds another 30 minutes or so on my test, as written (run with `-mp --max-workers 16` on a _slower_ instance type than you are using)

It is parallelized :-)    See where it is invoked.

WAG, but based upon a memory budget (3 columns, 12 bytes per row, 95% sparsity, 60K columns), so a WAG, but not a SWAG.

Correct - unfortunately IMHO. I think Paul deleted this capability in his refactoring.  I consider it a future (post-1.0) discussion, as the use cases are _very_ narrow.

Reply:
> It adds another 30 minutes

Great, that's acceptable (not hours).

> It is parallelized :-) See where it is invoked.

Indeed! Forest for the trees... And I should remember who I'm talking to here. 😄 

> I consider [specifying read attributes] a future (post-1.0) discussion, as the use cases are very narrow.

Yeah, this is about the only case I can imagine needing it. If performance was bad, was going to suggest diving down to the TileDB-Py API level for this. All good!